opencog/reasoning/pln/rules/evaluation-implication-rule.scm

```diff
@@ -0,0 +1,65 @@
+; =============================================================================
+; Evaluation Implication Rule
+;
+; AndLink
+;   EvaluationLink
+;       A
+;       B
+;   ImplicationLink
+;       A
+;       C
+; |-
+; EvaluationLink
+;   C
+;   B
+; -----------------------------------------------------------------------------
+(load "formulas.scm")
+
+(define pln-rule-evaluation-implication
+    (BindLink
+        (VariableList
+            (VariableNode "$A")
+            (VariableNode "$B")
+            (VariableNode "$C"))
+        (AndLink
+            (VariableNode "$A")
+            (VariableNode "$B")
+            (VariableNode "$C")
+            (EvaluationLink
+                (VariableNode "$A")
+                (VariableNode "$B"))
+            (ImplicationLink
+                (VariableNode "$A")
+                (VariableNode "$C")))
+        (ExecutionOutputLink
+            (GroundedSchemaNode "scm: pln-formula-evaluation-implication")
+            (ListLink
+                (VariableNode "$A")
+                (VariableNode "$B")
+                (VariableNode "$C")
+                (EvaluationLink
+                    (VariableNode "$A")
+                    (VariableNode "$B"))
+                (ImplicationLink
+                    (VariableNode "$A")
+                    (VariableNode "$C"))
+                (EvaluationLink
+                    (VariableNode "$C")
+                    (VariableNode "$B"))))))
+
+(define (pln-formula-evaluation-implication A B C AB AC CB)
+    (cog-set-tv!
+        CB
+        (stv
+            (simple-deduction-formula 
+                (cog-stv-strength B)
+                (cog-stv-strength A)
+                (cog-stv-strength C)
+                (cog-stv-strength AB)
+                (cog-stv-strength AC))
+            (min
+                (cog-stv-confidence B)
+                (cog-stv-confidence A)
+                (cog-stv-confidence C)
+                (cog-stv-confidence AC)
+                (* 0.9 (cog-stv-confidence AB))))))
```

Comment:
I think there are multiple discount factors, not just one.  Looking at the old python code it seems to be

`evalCB = 0.9 * 0.9 * (min (0.9 * evalAB), implAC, A, C)`

According to the comment on the python code (perhaps you should add that comment to here), it is from Chapter 10: Higher Order Extensional Inference.  It seems it is online at http://download.springer.com/static/pdf/392/chp%253A10.1007%252F978-0-387-76872-4_10.pdf?originUrl=http%3A%2F%2Flink.springer.com%2Fchapter%2F10.1007%2F978-0-387-76872-4_10&token2=exp=1439788690~acl=%2Fstatic%2Fpdf%2F392%2Fchp%25253A10.1007%25252F978-0-387-76872-4_10.pdf%3ForiginUrl%3Dhttp%253A%252F%252Flink.springer.com%252Fchapter%252F10.1007%252F978-0-387-76872-4_10*~hmac=ca35e8abf0f639b73dccf63f69e37962fd7bd29ab0d15e8291d6684ebb45fa7f and it also mentions the multiple discounts.